src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/SystemCommandLine/Utils.tt

```diff
@@ -445,10 +473,114 @@ public class OptionData
         bool hasCustomHandler = !string.IsNullOrEmpty(customHandler);
         if (hasCustomHandler)
         {
-            WriteLine($"{commandVarName}.SetHandler({customHandler});");
+            Write($"{commandVarName}.SetHandler({customHandler}");
+            Write(GenerateCommandHandlerFunctionArgs(node));
+            WriteLine(");");
         }
         PopIndent();
 
         return hasCustomHandler;
     } // GenerateSetCustomHandler
+
+    ValueTuple<string, string> GetVerbNounFormalNames(XElement verb, XElement noun = null)
+    {
+        string verbName = verb.Attribute(XName.Get("Name", ""))?.Value;
+        string verbFormalName = StringUtilities.GetFormalName(verbName);
+
+        string nounFormalName = string.Empty;
+        if (noun != null)
+        {
+            string nounName = noun.Attribute(XName.Get("Name", ""))?.Value;
+            nounFormalName = StringUtilities.GetFormalName(nounName);
+        }
+
+        return (verbFormalName, nounFormalName);
+    }
+
+    string GetArgsClassName(XElement verb, XElement noun = null)
+    {
+        (string verbFormalName, string nounFormalName) = GetVerbNounFormalNames(verb, noun);
+
+        return $"{verbFormalName}{nounFormalName}Args";
+    }
+
+    string GetCustomBinderClassName(XElement verb, XElement noun = null)
+    {
+        (string verbFormalName, string nounFormalName) = GetVerbNounFormalNames(verb, noun);
+
+        return $"{verbFormalName}{nounFormalName}CustomBinder";
+    }
```

Comment:
Removed duplicate code but, I will keep GetCustomBinderClassName and GetArgsClassName methods because I will need them later when generating custom args class.